cpp/game_main.cpp

```diff
@@ -346,16 +346,27 @@ bool GameMain::on_input(SDL_Event *e) {
 			if (this->building_placement) {
 
 				// confirm building placement with left click
-				Command cmd(this->players[0], this->datamanager.get_producer_index(this->editor_current_building), mousepos_phys3);
-				cmd.set_ability(ability_type::build);
-				selection.all_invoke(cmd);
+				// first create foundation using the producer
+				auto player = this->selection.owner();
+				if (player) {
+					auto container = &this->placed_units;
+					auto building_type = this->datamanager.get_type_index(this->editor_current_building);
+					auto new_building = container->new_unit(*building_type, *player, mousepos_phys3);
+
+					// task all selected villagers to build
+					if (new_building.is_valid()) {
+						Command cmd(*player, new_building.get());
+						cmd.set_ability(ability_type::build);
+						this->selection.all_invoke(cmd);
+					}	
```

Comment:
http://sublimetexttips.com/show-whitespace-sublime-text/

You can control when Sublime shows white space with the draw_white_space setting. The default value is selection, but you can override that with all to always show whitespace or none to never show it.

To change your whitespace preferences, run Preferences: Settings – User in the command palette, then add the draw_white_space setting:

{
    "draw_white_space": "all"
}